Automattic/woocommerce-payments
# RPP / Add 100% unit test coverage requirement for src

#### Changes proposed in this Pull Request

Allow the `src` directory's unit coverage to be checked separately from `includes`.

1. There is a new `test:php-coverage-src` npm command, which only checks `src`, and is significantly faster than the full `npm run test:php-coverage` command.
2. A new "Code coverage (src) / Code coverage" workflow in GitHub which fails if coverage is less than 100%.

How this was achieved:

- Introduced a custom `phpunit-src.xml.dist` to allow the selection of only the `src` directory. Otherwise it is the same as `phpunit.xml.dist`.
- `bin/phpunit.sh` does not run by with `-c phpunit.xml.dist` by default anymore, instead allowing other commands to define it.
- Introduced 3 new files:
    1. `bin/check-src-test-coverage.sh` for the local command. Identical to `bin/check-test-coverage.sh`, but with the new XML config.
    2. `bin/run-ci-tests-check-src-coverage.bash` for the CI job, almost identical to the non-`src` one as well.
    3. `.github/workflows/src-coverage.yml` is the definition of the GitHub workflow.

#### Testing instructions

1. Run both `npm run test:php-coverage` and `npm run test:php-coverage-src` locally to make sure they work.
2. All CI checks should pass in GitHub.

-------------------
